### DISCUSSION TITLE
[Help]: Is there a way to receive events or toggle classnames when a slide starts and completes its movement?

### DISCUSSION THREAD

**Discussion by asdf99245** [davidjerleke/embla-carousel#1093] [Help] [answered]
### Summary

I'd like to add a specific classname to slides when they start moving, and remove the classname when the movement is completed. Is there a way to implement this currently?

I tried adding the classname on the `select` event and removing it on the `settle` event, but the `settle` event occurs only after the carousel has completely stopped playing.

> For example, when using the autoplay feature, the 'settle' event didn't occur until the autoplay stopped.

What I'm looking for is a way to perform some action at the moment a slide starts moving and when it finishes moving, for each individual slide movement.

Thank you for providing such a great library! 👍 

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [X] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment]
Hi @asdf99245,
Thanks for your question. If the `settle` event doesn’t fire, that probably means that your `delay` for the autoplay plugin is too short. Basically, it triggers the next slide change before the carousel has settled. 

So try increasing your `delay` option and see if it solves the problem.

**Comment by asdf99245** [comment]
@davidjerleke Thank you for answering 😃 

So Here is the sample demo!
https://codesandbox.io/p/sandbox/drgmlk

I need to set the autoplay delay to a minimum of 1000ms, but I've noticed that the settle event only occurs from a 2000ms delay onwards.

Additionally, I've observed that when moving with scrollTo outside of autoplay, it moves quickly and then autoplays again, causing the settle event not to occur.

Also, it appears that the settle event doesn't happen immediately after the movement, but with a slight delay!
What I want to do is manipulate the slides at the start and end of each movement, regardless of the delay, every time a slide moves. Is there currently a way to achieve this? It would be really great if there is!

Thank you 👍

**Reply by davidjerleke** [reply, marked as answer]
@asdf99245 you can use the select event when the carousel selects a new target (starts animating), and use the snippet to in the following discussion to add a settle threshold:

- https://github.com/davidjerleke/embla-carousel/discussions/738#discussioncomment-8404744

Regarding this:

> Additionally, I've observed that when moving with scrollTo outside of autoplay, it moves quickly and then autoplays again, causing the settle event not to occur.

This is expected. If you check the [Autoplay example](https://www.embla-carousel.com/examples/predefined/#autoplay) in the docs, you will see that you need to call the autoplay reset method when calling scrollTo to avoid this.

**Reply by asdf99245** [reply]
Okay, I understand.
It works perfectly! Thank you for providing such an excellent library! 🥇
